src/Assert.php

```diff
@@ -466,6 +549,9 @@ public static function false($value, $message = '')
         }
     }
 
+    /**
+     * @psalm-assert string $value
+     */
     public static function ip($value, $message = '')
     {
         if (false === filter_var($value, FILTER_VALIDATE_IP)) {
```

Comment:
Value is not guaranteed to be a string after this point. The documentation on [`filter_var`](https://www.php.net/manual/en/function.filter-var.php) mentions the following:
> Value to filter. Note that scalar values are converted to string internally before they are filtered.

for example: https://3v4l.org/2mpMA